### PR TITLE
Add build flag for fast math

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

**Comment by pfultz2** [comment]
The python docs need to be updated.

File: test/gpu/ops_test.cpp [hunk 1]
```diff
@@ -2584,6 +2585,49 @@ void manual_identity()
...
+    options.fast_math = false;
+    p.compile(migraphx::gpu::target{}, options);
+    bool found_gelu_new = false;
+    for(auto ins : iterator_for(p))
+    {
+        if(ins->name() == "gpu::gelu_new")
+            found_gelu_new = true;
+    }
+    CHECK(found_gelu_new);
+}
```
Comment (pfultz2) [inline test/gpu/ops_test.cpp:2629]: This should be put in a different cpp file.

Comment (pfultz2) [reply test/gpu/ops_test.cpp:2629]: Also, there should be two tests, one with it enabled and another with it disabled.

Comment (kahmed10) [reply test/gpu/ops_test.cpp]: fixed

File: test/gpu/fast_math.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            found_gelu = true;
+    }
+    CHECK(found_gelu);
+}
+
+TEST_CASE(enable_fast_gelu)
+{
+    migraphx::program p = create_gelu();
+    p.compile(migraphx::gpu::target{});
+    check_gelu_version(p, "gpu::gelu");
```
Comment (pfultz2) [inline test/gpu/fast_math.cpp:56]: This could be written with `any_of`:

```cpp
CHECK(any_of(p, [&](auto&& i) { return i.name() == "gpu::gelu"; });
```

**Review by scxiao — APPROVED** [review]

File: test/gpu/fast_math.cpp (hunk 2 above)
Comment (kahmed10) [reply test/gpu/fast_math.cpp]: fixed

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**